src/Composer/Plugin/PluginManager.php

```diff
@@ -462,10 +466,28 @@ private function loadRepository(RepositoryInterface $repo, bool $isGlobalRepo):
         }
 
         $sortedPackages = PackageSorter::sortPackages($packages, $weights);
+        if (!$isGlobalRepo) {
+            $requiredPackages = RepositoryUtils::filterRequiredPackages($packages, $rootPackage, true);
+        }
+
         foreach ($sortedPackages as $package) {
             if (!($package instanceof CompletePackage)) {
                 continue;
             }
+
+            if (!in_array($package->getType(), ['composer-plugin', 'composer-installer'],true)) {
```

Comment:
```suggestion
            if (!in_array($package->getType(), ['composer-plugin', 'composer-installer'], true)) {
```
and here's a spacing fix ;-P